relay-general/src/protocol/contexts/os.rs

```diff
@@ -73,14 +73,33 @@ impl FromUserAgentInfo for OsContext {
 
     fn parse_user_agent(user_agent: &str) -> Option<Self> {
         let os = parse_os(user_agent);
+        let mut version = get_version(&os.major, &os.minor, &os.patch);
 
         if !is_known(&os.family) {
             return None;
         }
 
+        let name = os.family.into_owned();
+
+        // Since user-agent strings freeze the OS-version at windows 10 and mac os 10.15.7,
+        // we will indicate that the version may in reality be higher.
+        if name == "Windows" {
+            if let Some(v) = version.as_mut() {
+                if v == "10" {
+                    v.insert_str(0, ">=");
```

Comment:
I figured doing this instead of `*v = ">=10".to_owned()' would prevent a heap allocation. Let me know if this is wrong.

Reply:
Sounds right to me.